pkg/controller/install/certresources_test.go

```diff
@@ -167,7 +169,35 @@ func TestInstallCertRequirementsForDeployment(t *testing.T) {
 						Selector: selector(t, "test=label").MatchLabels,
 					},
 				}
-				mockOpClient.EXPECT().CreateService(&service).Return(&service, nil)
+
+				portsApplyConfig := []*corev1ac.ServicePortApplyConfiguration{}
+				for _, p := range args.ports {
+					ac := corev1ac.ServicePort().
+						WithName(p.Name).
+						WithPort(p.Port).
+						WithTargetPort(p.TargetPort)
+					portsApplyConfig = append(portsApplyConfig, ac)
+				}
+				ownerRef := ownerutil.NonBlockingOwner(&v1alpha1.ClusterServiceVersion{})
+				ownerRefAC := metav1ac.OwnerReference().
+					WithAPIVersion(ownerRef.APIVersion).
+					WithKind(ownerRef.Kind).
+					WithUID(ownerRef.UID).
+					WithName(ownerRef.Name)
+				if ownerRef.BlockOwnerDeletion != nil {
+					ownerRefAC.WithBlockOwnerDeletion(*ownerRef.BlockOwnerDeletion)
+				}
+				if ownerRef.Controller != nil {
+					ownerRefAC.WithController(*ownerRef.Controller)
+				}
+				svcApplyConfig := corev1ac.Service(service.Name, service.Namespace).
+					WithSpec(corev1ac.ServiceSpec().
+						WithPorts(portsApplyConfig...).
+						WithSelector(selector(t, "test=label").MatchLabels)).
+					WithOwnerReferences(ownerRefAC).
+					WithLabels(map[string]string{OLMManagedLabelKey: OLMManagedLabelValue})
+
+				mockOpClient.EXPECT().ApplyService(svcApplyConfig, metav1.ApplyOptions{Force: true, FieldManager: "olm.install"}).Return(&service, nil)
```

Comment:
I don't know if the mock client testing bits support apply :|

Reply:
Fortunately, these calls do work properly.